src/si/amp.py

```diff
@@ -54,7 +56,11 @@ def __init__(self, key, domains, **options):
             url = '/test/update_from_spa/' + self.key + "?session_life_time=%s" % self._session_lifetime
             conn.request('GET', url)
             response = conn.getresponse()
-            if response.getcode() != 200:
+            if sys.version_info[0] == 3:
```

Comment:
you might want to create a get_response_code function which hides all of this ugliness